client/components/data/query-users-suggestions/index.jsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingUserSuggestions as isRequesting } from 'state/users/suggestions/selectors';
+import { requestUserSuggestions } from 'state/users/suggestions/actions';
+
+class QueryUsersSuggestions extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId === nextProps.siteId ) {
+			return;
+		}
+
+		this.request( nextProps );
+	}
+
+	request( props ) {
+		if ( props.isRequesting ) {
+			return;
+		}
+
+		props.requestUserSuggestions( props.siteId );
```

Comment:
Being able to handle falsey props make the QueryComponents a bit easier to use. For example if we pull current siteId from redux state, this might not be guaranteed to be populated during the first render.
So by handling falsey values we write this as:
```javascript
<QueryUsersSuggestions siteId={ this.props.siteId } />
```
vs
```javascript
this.props.siteId && <QueryUsersSuggestions siteId={ this.props.siteId } />
```